### PR TITLE
Full auto routine

### PR DESCRIPTION
**Pull request by asherbolotski**
this is the code for the full auto routines, just needs the grabber command and the arm positioning commands and then it is good to go

### CODE REVIEW

File: smokey/src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -73,6 +73,10 @@ public final class Constants {
     public static final double TrajectoryMaxAcceleration = 1;
     public static final double TrajectoryMaxVelocity = 1;
 
+    //*************************************** 
+    //For auto constants
+    public static final double snoutDepth = 5.25;
```
Comment (josieammer) [inline smokey/src/main/java/frc/robot/Constants.java:78]: Call this snoutDepthMeters and do the Units.inchesToMeters conversion here.

File: smokey/src/main/java/frc/robot/commands/ManipulatePickerCommand.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+// Intent: Forms a command to open or close the picker.
+// ************************************************************
+
+// ʕ •ᴥ•ʔ ʕ•ᴥ•  ʔ ʕ  •ᴥ•ʔ ʕ •`ᴥ´•ʔ ʕ° •° ʔ ʕ •ᴥ•ʔ ʕ•ᴥ•  ʔ ʕ  •ᴥ•ʔ ʕ •`ᴥ´•ʔ ʕ° •° ʔ 
+
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+public class ManipulatePickerCommand extends CommandBase {
```
Comment (josieammer) [inline smokey/src/main/java/frc/robot/commands/ManipulatePickerCommand.java:15]: Mike has a PR for grabber code.  You should incorporate here before merging.

File: smokey/src/main/java/frc/robot/commands/MoveArmToScoring.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+// Intent: Forms a command to bring the arm to an inputed scoring position.
+// ************************************************************
+
+// ʕ •ᴥ•ʔ ʕ•ᴥ•  ʔ ʕ  •ᴥ•ʔ ʕ •`ᴥ´•ʔ ʕ° •° ʔ ʕ •ᴥ•ʔ ʕ•ᴥ•  ʔ ʕ  •ᴥ•ʔ ʕ •`ᴥ´•ʔ ʕ° •° ʔ 
+
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+public class MoveArmToScoring extends CommandBase {
```
Comment (josieammer) [inline smokey/src/main/java/frc/robot/commands/MoveArmToScoring.java:15]: Naher is working on a class that will replace this one.  You should incorporate before merging.

File: smokey/src/main/java/frc/robot/commands/MoveArmToStowed.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+// Intent: Forms a command to bring the arm to an inputed scoring position.
+// ************************************************************
+
+// ʕ •ᴥ•ʔ ʕ•ᴥ•  ʔ ʕ  •ᴥ•ʔ ʕ •`ᴥ´•ʔ ʕ° •° ʔ ʕ •ᴥ•ʔ ʕ•ᴥ•  ʔ ʕ  •ᴥ•ʔ ʕ •`ᴥ´•ʔ ʕ° •° ʔ 
+
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+public class MoveArmToStowed extends CommandBase {
```
Comment (josieammer) [inline smokey/src/main/java/frc/robot/commands/MoveArmToStowed.java:15]: Naher is working on a class that will replace this one. You should incorporate before merging.

File: smokey/src/main/java/frc/robot/control/AutonomousChooser.java [hunk 5]
```diff
@@ -16,25 +16,34 @@
...
 public class AutonomousChooser {
     private SubsystemCollection subsystems;
-    private final SendableChooser<AutonomousMode> autonomousModeChooser = new SendableChooser<>();
+    private final SendableChooser<AutonomousPath> AutonomousPathChooser = new SendableChooser<>();
+    private final SendableChooser<AutonomousBalance> balanceChooser = new SendableChooser<>();
+    private final SendableChooser<ScoringPosition> scoreHeight = new SendableChooser<>();
     private Trajectories trajectories;
-
+    
+    private Transform2d IntoNodeTransform = new Transform2d(new Translation2d(Units.inchesToMeters(Constants.snoutDepth * -1), 0), new Rotation2d(0.0));
```
Comment (josieammer) [inline smokey/src/main/java/frc/robot/control/AutonomousChooser.java:45]: variable name should have lower case first letter: intoNodeTransform.

File: smokey/src/main/java/frc/robot/control/AutonomousChooser.java [hunk 6]
```diff
@@ -16,25 +16,34 @@
...
+    private Transform2d OutOfNodeTransform = new Transform2d(new Translation2d(Units.inchesToMeters(Constants.snoutDepth), 0), new Rotation2d(0.0));
```
Comment (josieammer) [inline smokey/src/main/java/frc/robot/control/AutonomousChooser.java:46]: same comment as above.

File: smokey/src/main/java/frc/robot/control/AutonomousChooser.java [hunk 7]
```diff
@@ -45,100 +54,87 @@ public AutonomousChooser(SubsystemCollection subsystems) {
...
-        command.addCommands(driveSegment(0, 0, 0.5, 2.0));
-
+        command.addCommands(new InstantCommand(() -> subsystems.getDriveTrainSubsystem().setRobotPosition(NodePosition)));
+        command.addCommands(new DriveToPointCommand(subsystems.getDriveTrainSubsystem(), subsystems.getDriveTrainSubsystem().getRobotPosition().plus(IntoNodeTransform)));
+        command.addCommands(new MoveArmToScoring(scoreHeight.getSelected()));
+        command.addCommands(new ManipulatePickerCommand(false));
+        command.addCommands(new DriveToPointCommand(subsystems.getDriveTrainSubsystem(), subsystems.getDriveTrainSubsystem().getRobotPosition().plus(OutOfNodeTransform)));
+        command.addCommands(new MoveArmToStowed());
+        command.addCommands(new DriveTrajectoryCommand(subsystems.getDriveTrainSubsystem(), Trajectory));
+        command.addCommands(new InstantCommand(() -> subsystems.getDriveTrainSubsystem().setRobotPosition(trajectories.End)));
```
Comment (josieammer) [inline smokey/src/main/java/frc/robot/control/AutonomousChooser.java:97]: remove this line.

File: smokey/src/main/java/frc/robot/control/AutonomousChooser.java [hunk 8]
```diff
@@ -45,100 +54,87 @@ public AutonomousChooser(SubsystemCollection subsystems) {
...
-     * @return command
+     * Builds a command list for the balance routine, or will not if we toggle it so
+     * @param DoBalance the enum for wether the balance routine should be run
+     * @return
      */
-    public Command getTestAutoBackward() {
+    public Command getBalanceRoutine (Enum DoBalance){
         SequentialCommandGroup command = new SequentialCommandGroup();
-
         resetRobotPose(command);
```
Comment (josieammer) [inline smokey/src/main/java/frc/robot/control/AutonomousChooser.java:108]: remove this line.

File: smokey/src/main/java/frc/robot/control/AutonomousChooser.java [hunk 9]
```diff
@@ -45,100 +54,87 @@ public AutonomousChooser(SubsystemCollection subsystems) {
...
 
-    /**
-     * A method to get the create a DriveToPointCommand using the current position as a starting point
-     * @return command
-     */
-    public Command getTestAutoToPosition() {
+    public Command getLeftRoutine(){
         SequentialCommandGroup command = new SequentialCommandGroup();
-
         resetRobotPose(command);
```
Comment (josieammer) [inline smokey/src/main/java/frc/robot/control/AutonomousChooser.java:118]: remove this line (it's already done in the getAutoRoutine

File: smokey/src/main/java/frc/robot/control/AutonomousChooser.java [hunk 10]
```diff
@@ -45,100 +54,87 @@ public AutonomousChooser(SubsystemCollection subsystems) {
...
+        command.addCommands(getAutoRoutine(trajectories.Node1Position, trajectories.LeftTrajectory));
+        command.addCommands(getBalanceRoutine(balanceChooser.getSelected()));
         return command;
-    }    
+    }
 
-    public Command getBlueUpAndOver() {
+    public Command getRightRoutine(){
         SequentialCommandGroup command = new SequentialCommandGroup();
         resetRobotPose(command);
```
Comment (josieammer) [inline smokey/src/main/java/frc/robot/control/AutonomousChooser.java:126]: remove this line

File: smokey/src/main/java/frc/robot/control/AutonomousChooser.java [hunk 11]
```diff
@@ -45,100 +54,87 @@ public AutonomousChooser(SubsystemCollection subsystems) {
...
-        resetRobotPose(command);
-        command.addCommands(new InstantCommand(() -> subsystems.getDriveTrainSubsystem().setRobotPosition(trajectories.BluStart)));
-        command.addCommands(new DriveTrajectoryCommand(subsystems.getDriveTrainSubsystem(), trajectories.BluDownAndUnderTrajectory));
-        return command;
-    }   
 
-    public Command getBluAcrossRamp() {
+    public Command getMiddleRoutine(){
         SequentialCommandGroup command = new SequentialCommandGroup();
         resetRobotPose(command);
```
Comment (josieammer) [inline smokey/src/main/java/frc/robot/control/AutonomousChooser.java:134]: remove this line.

File: smokey/src/main/java/frc/robot/control/Trajectories.java [hunk 12]
```diff
@@ -11,74 +11,50 @@
...
-    public static Trajectory BluOntoRampTrajectory;
-    public static Trajectory RedUpAndOverTrajectory;
-    public static Trajectory RedDownAndUnderTrajectory;
-    public static Trajectory RedAcrossRampTrajectory;
-    public static Trajectory RedOntoRampTrajectory;
-
+    public Pose2d Node1Position;
+    public Pose2d Node5Position;
+    public Pose2d Node9Position;
+    public Pose2d End;
```
Comment (josieammer) [inline smokey/src/main/java/frc/robot/control/Trajectories.java:17]: rename this something better, like outOfCommunityPosition

File: smokey/src/main/java/frc/robot/control/Trajectories.java [hunk 13]
```diff
@@ -11,74 +11,50 @@
...
-        Pose2d BluEnd = new Pose2d(5.81, 2.748, Rotation2d.fromDegrees(90));
-        Pose2d BluRamp = new Pose2d(3.922, 2.748, Rotation2d.fromDegrees(90));
- 
-        ArrayList<Translation2d> BluUpAndOverWaypoints = new ArrayList<Translation2d>();
-        BluUpAndOverWaypoints.add(new Translation2d(2.1, 4.67));
-        BluUpAndOverWaypoints.add(new Translation2d(3.7, 4.67));
-        BluUpAndOverWaypoints.add(new Translation2d(5.3, 4.67));
-        System.out.println(">>>>> Generating Blue up and over");
-        this.BluUpAndOverTrajectory = SwerveTrajectoryGenerator.generateTrajectory(BluStart, BluUpAndOverWaypoints, BluEnd, config);
+        this.Node1Position = new Pose2d(0, 0, Rotation2d.fromDegrees(180));
```
Comment (josieammer) [inline smokey/src/main/java/frc/robot/control/Trajectories.java:28]: add in the correct positions from the spreadsheet

File: smokey/src/main/java/frc/robot/control/Trajectories.java [hunk 14]
```diff
@@ -11,74 +11,50 @@
...
+        this.Node5Position = new Pose2d(1.678, 2.748, Rotation2d.fromDegrees(180));
```
Comment (josieammer) [inline smokey/src/main/java/frc/robot/control/Trajectories.java:29]: add in the correct positions from the spreadsheet

File: smokey/src/main/java/frc/robot/control/Trajectories.java [hunk 15]
```diff
@@ -11,74 +11,50 @@
...
+        this.Node9Position = new Pose2d(0, 0, Rotation2d.fromDegrees(180));
```
Comment (josieammer) [inline smokey/src/main/java/frc/robot/control/Trajectories.java:30]: add in the correct positions from the spreadsheet

**Review by josieammer — CHANGES_REQUESTED** [review]

File: smokey/src/main/java/frc/robot/Constants.java (hunk 1 above)
Comment (asherbolotski) [reply smokey/src/main/java/frc/robot/Constants.java:78]: done

File: smokey/src/main/java/frc/robot/control/AutonomousChooser.java (hunk 5 above)
Comment (asherbolotski) [reply smokey/src/main/java/frc/robot/control/AutonomousChooser.java:45]: done

File: smokey/src/main/java/frc/robot/control/AutonomousChooser.java (hunk 6 above)
Comment (asherbolotski) [reply smokey/src/main/java/frc/robot/control/AutonomousChooser.java:46]: yup

File: smokey/src/main/java/frc/robot/Constants.java [hunk 16]
```diff
@@ -73,6 +74,10 @@ public final class Constants {
     public static final double TrajectoryMaxAcceleration = 1;
     public static final double TrajectoryMaxVelocity = 1;
 
+    //*************************************** 
+    //For auto constants
+    public static final double snoutDepth = Units.inchesToMeters(5.25);
```
Comment (josieammer) [inline smokey/src/main/java/frc/robot/Constants.java:79]: should be called snoutDepthMeters

Comment (josieammer) [reply smokey/src/main/java/frc/robot/Constants.java:88]: fixed.

**Review by josieammer — APPROVED** [review]
Reviewed and well-tested on robot. Pushing to main.

**Pull request merged**